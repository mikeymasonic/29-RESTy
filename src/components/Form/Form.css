.Form {
  margin: auto;
  width: 25rem;
  background-color: purple;
  color: whitesmoke;
  padding: 20px;
}

h3 {
  text-align: center;
}

p {
  padding: 1rem 2rem;
  line-height: 1.5rem;
}