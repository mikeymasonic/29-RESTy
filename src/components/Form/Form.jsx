import React from 'react';
import styles from './Form.css';

const Form = () => (
  <div className={styles.Form}>
    <form>
      <label>API Url:<input type="text"></input></label> 
      <div>
        <label><input type="radio" name="method" value="get" />GET</label>
        <label><input type="radio" name="method" value="post" />POST</label>
        <label><input type="radio" name="method" value="put" />PUT</label>
        <label><input type="radio" name="method" value="patch" />PATCH</label>
        <label><input type="radio" name="method" value="delete" />DELETE</label>
      </div>
      <textarea placeholder="Body"></textarea>
      <button type="submit">Submit</button>
    </form>
  </div>
);

export default Form;
