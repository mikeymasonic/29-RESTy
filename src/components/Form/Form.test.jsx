import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form />);
    expect(wrapper).toMatchSnapshot();
  });
});
