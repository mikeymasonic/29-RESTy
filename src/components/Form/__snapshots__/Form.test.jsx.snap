// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<div
  className="Form"
>
  <form>
    <label>
      URL:
      <input
        name="url"
        type="text"
      />
    </label>
    <div>
      <label>
        <input
          name="method"
          type="radio"
          value="GET"
        />
        GET
      </label>
      <label>
        <input
          name="method"
          type="radio"
          value="POST"
        />
        POST
      </label>
      <label>
        <input
          name="method"
          type="radio"
          value="PUT"
        />
        PUT
      </label>
      <label>
        <input
          name="method"
          type="radio"
          value="PATCH"
        />
        PATCH
      </label>
      <label>
        <input
          name="method"
          type="radio"
          value="DELETE"
        />
        DELETE
      </label>
    </div>
    <textarea
      name="body"
      placeholder="Body"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
